x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/retry/BaseResponseHandler.java

```diff
@@ -95,7 +95,7 @@ public void validateResponse(
 
     protected abstract void checkForFailureStatusCode(Request request, HttpResult result);
 
-    private void checkForErrorObject(Request request, HttpResult result) {
+    protected void checkForErrorObject(Request request, HttpResult result) {
```

Comment:
We need to be able to override this if we want to use a different error class. This PR adds `UnifiedChatCompletionErrorResponse` which pulls the common fields for constructing a `UnifiedChatCompletionException` into a single place.